docs/sources/style-guide/write-for-developers/index.md

```diff
@@ -0,0 +1,153 @@
+- When the name of a code element is singular, don't pluralize it. Instead, follow it by plural noun. For example, write "`MyEvent` objects" instead of `MyEvents`.
+- If the element performs an action, start the first sentence of the description with an action verb. 
+
+## Code examples
+
+Readers of documentation typically skim through it to find code samples they can copy and paste and run as is.
+Because of this and whenever possible, provide production-ready examples.
+
+However, it isn't necessary for every code example to be runnable in production.  Some code examples are written to illustrate a point so that the developer can learn how to do something similar on their own.
+
+When providing an example, give a written description. You can put it either in the body of the document or as explanatory comments within the example code.
+
+Remember the basic rule: explain _why_ your code does what it does, rather than describe _what_ it does.
+For an in-depth, external resource about writing developer documentation, see [_Docs for Developers_](https://docsfordevelopers.com/).
+
+### Formatting code examples
+
+Here are guidelines to follow when formatting code examples.
+
+- Use spaces, not tabs. 
+- Follow Grafana's accepted [coding style guidelines](https://github.com/grafana/grafana/blob/main/contribute/style-guides/).
+- Wrap lines at 80 characters.
+- When omitting code, use three dots (...). Don't use the ellipsis character (…).
+
+Introduce each code sample with a sentence or paragraph to establish its context. End the introduction with a colon if it immediately precedes the sample, or a period if it doesn't. 
+
+### Paths, filenames, and URLs
+
+Many types of information belong in fixed-width font. Among these are path names, file names, directories, or folders. However, don't format domain names or URLs as code if you intend the user to follow the link.
+
+Although there are times when you might want to make a URL clickable, it is better to follow the guidelines for [References]({{< relref "../../writing-guide/references/" >}}).
+
+## Command lines
+
+Use the following conventions when you include commands on the command line in technical content.
+
+- Do not assume everyone is using Linux. Make sure instructions include enough information for Windows and Mac users to successfully complete procedures.
+
+- Do not add `$` before commands. Make it easy for users to copy and paste commands.
+
+  - **Right:** `sudo yum install grafana`
+  - **Wrong:** `$ sudo yum install grafana`
+
+- Include `sudo` before commands that require `sudo` to work.
+
+For terminal examples and configurations, use a `bash` code block:
+
+```bash
+sudo yum install grafana
+```
+
+If your command-line instructions include a combination of input and output lines, use separate code blocks for input and output, and use a `console` code block for the output.
```

Comment:
I'm finding it difficult to follow this conversation in a couple of ways. @JStickler @osg-grafana. Although I personally would prefer to add a code example to the Markdown syntax page, if we are going to add it here, could someone please supply it for me? I think isn't it just

```console
comand command command
```

@jdbaldry  seems to have confirmed that a certain syntax is correct, but I can't tell from the context what he is referring to.

Reply:
Note Julie raised the same issue below with bash (adding this for my own reference because I resolved the comment too soon).

Reply:
1 - To install [A thing] run the following commands:

```
```bash
command #1
```
```bash
output
```
```bash
command #2
```
```bash
output
```
```bash
command #3
```
```bash
success message
```
```

Reply:
Ugh, I am failing today at providing a code sample that will show in the comments the way I want it to.  =(

Reply:
But the general idea is:
1 - To install [A thing] run the following commands:
Code block with command #1 
Separate code block with expected output /success message
Code block with command #2 
Separate code block with expected output /success message
Code block with command #3 
Separate code block with expected output /success message
2 - To configure [a thing]...

Reply:
Hi Julie, Sam Boyer reviewed this section and I updated the PR today to accomodate his suggestions. I'm not sure I succeeded at getting the code examaples to format correctly, so you're welcome to take a look at the updated PR and re-review.

Reply:
Resolving for now, we can modify later if necessary.